pkg/topology/topology.go

```diff
@@ -39,6 +41,27 @@ const (
 	Balanced
 )
 
-// LinstorNodeKey refers to a node running the LINSTOR csi node service
-// and the linstor Satellite and is therefore capabile of hosting LINSTOR volumes.
-const LinstorNodeKey = "linbit.com/hostname"
+const (
+	// LinstorNodeKey refers to a node running the LINSTOR csi node service
+	// and the linstor Satellite and is therefore capable of hosting LINSTOR volumes.
+	LinstorNodeKey = "linbit.com/hostname"
+
+	// LinstorStoragePoolKeyPrefix is the prefix used when specifying the available storage
+	// pools on a node via CSI topology keys.
+	LinstorStoragePoolKeyPrefix = "linbit.com"
+
+	// The value assigned to the storage pool label, given that the node has access to the storage pool.
+	LinstorStoragePoolValue = "true"
+)
+
+// Converts the storage pool name into a CSI Topology compatible label.
+//
+// There is an upper limit on the length of these keys (63 chars for prefix + 63 chars for the key) as per CSI Spec.
+// LINSTOR enforces a stricter limit of 48 characters for storage pools, so this should not be an issue
+func ToStoragePoolLabel(storagePoolName string) string {
```

Comment:
sorry for being picky on that, but I'd like to have the check on the final generated string. something like:
```go
label := fmt.Sprintf("%s/sp-%s", LinstorStoragePoolKeyPrefix, storagePoolName)
if len(label) > 63+1+63 { // should be impossible
        panic(....)
}```

Reply:
Thinking about this again, this approach is essentially the same as not checking at all. In this case our pod panics, without the check the registrar plugin stops with a helpful error message: 
> `Invalid value: "linbit.com/sp-....": name part must be no more than 63 characters`
So I'd be in favour of not checking at all.

Reply:
yes, fine with me, just keep what we already documented, that it is impossible, and that it would fail anyways.